From 0b9c20f5e8a1d16f8995220ad78bb6ba07e425c2 Mon Sep 17 00:00:00 2001
From: ChonDoit <thphantomblog@gmail.com>
Date: Sun, 27 Oct 2024 21:52:42 +0100
Subject: [PATCH 5/9] Make status bar paddings overridables

with
persist.sys.phh.status_bar_padding_top
persist.sys.phh.status_bar_padding_start
persist.sys.phh.status_bar_padding_end
---
 .../phone/KeyguardStatusBarView.java          |  7 +++--
 .../statusbar/phone/PhoneStatusBarView.java   | 26 +++++++++++++------
 2 files changed, 23 insertions(+), 10 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardStatusBarView.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardStatusBarView.java
index 0b8d7f531..6b56dceaa 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardStatusBarView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/KeyguardStatusBarView.java
@@ -28,6 +28,7 @@ import android.graphics.Color;
 import android.graphics.Insets;
 import android.graphics.Rect;
 import android.graphics.drawable.Drawable;
+import android.os.SystemProperties;
 import android.os.Trace;
 import android.util.AttributeSet;
 import android.util.TypedValue;
@@ -158,7 +159,7 @@ public class KeyguardStatusBarView extends RelativeLayout {
         // mStatusIconArea
         mStatusIconArea.setPaddingRelative(
                 mStatusIconArea.getPaddingStart(),
-                getResources().getDimensionPixelSize(R.dimen.status_bar_padding_top),
+                mStatusIconArea.getPaddingTop(),
                 mStatusIconArea.getPaddingEnd(),
                 mStatusIconArea.getPaddingBottom()
         );
@@ -207,7 +208,9 @@ public class KeyguardStatusBarView extends RelativeLayout {
         Resources res = getResources();
         mSystemIconsSwitcherHiddenExpandedMargin = res.getDimensionPixelSize(
                 R.dimen.system_icons_switcher_hidden_expanded_margin);
-        mStatusBarPaddingEnd = res.getDimensionPixelSize(
+        mStatusBarPaddingEnd = SystemProperties.getInt("persist.sys.phh.status_bar_padding_end", -1);
+    	if(mStatusBarPaddingEnd == -1)
+            mStatusBarPaddingEnd = res.getDimensionPixelSize(
                 R.dimen.status_bar_padding_end);
         mMinDotWidth = res.getDimensionPixelSize(
                 R.dimen.ongoing_appops_dot_min_padding);
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarView.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarView.java
index 18604bfb6..b13c97001 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/PhoneStatusBarView.java
@@ -28,6 +28,7 @@ import android.graphics.Rect;
 import android.inputmethodservice.InputMethodService;
 import android.os.IBinder;
 import android.os.RemoteException;
+import android.os.SystemProperties;
 import android.util.AttributeSet;
 import android.util.Log;
 import android.view.ContextThemeWrapper;
@@ -354,22 +355,31 @@ public class PhoneStatusBarView extends FrameLayout implements Callbacks {
     }
 
     private void updatePaddings() {
-        int statusBarPaddingStart = getResources().getDimensionPixelSize(
-                R.dimen.status_bar_padding_start);
+        int statusBarPaddingTop = SystemProperties.getInt("persist.sys.phh.status_bar_padding_top", -1);
+        if(statusBarPaddingTop == -1)
+            statusBarPaddingTop = getResources().getDimensionPixelSize(R.dimen.status_bar_padding_top);
+
+        int statusBarPaddingStart = SystemProperties.getInt("persist.sys.phh.status_bar_padding_start", -1);
+        if(statusBarPaddingStart == -1)
+            statusBarPaddingStart = getResources().getDimensionPixelSize(R.dimen.status_bar_padding_start);
+
+        int statusBarPaddingEnd = SystemProperties.getInt("persist.sys.phh.status_bar_padding_end", -1);
+        if(statusBarPaddingEnd == -1)
+            statusBarPaddingEnd = getResources().getDimensionPixelSize(R.dimen.status_bar_padding_end);
 
         mStatusBarContents.setPaddingRelative(
-                statusBarPaddingStart,
-                getResources().getDimensionPixelSize(R.dimen.status_bar_padding_top),
-                getResources().getDimensionPixelSize(R.dimen.status_bar_padding_end),
+                (int) statusBarPaddingStart,
+                (int) statusBarPaddingTop,
+                (int) statusBarPaddingEnd,
                 0);
 
         findViewById(R.id.notification_lights_out)
                 .setPaddingRelative(0, statusBarPaddingStart, 0, 0);
 
         findViewById(R.id.system_icons).setPaddingRelative(
-                getResources().getDimensionPixelSize(R.dimen.status_bar_icons_padding_start),
-                getResources().getDimensionPixelSize(R.dimen.status_bar_icons_padding_top),
-                getResources().getDimensionPixelSize(R.dimen.status_bar_icons_padding_end),
+                (int) statusBarPaddingStart,
+                (int) statusBarPaddingTop,
+                (int) statusBarPaddingEnd,
                 getResources().getDimensionPixelSize(R.dimen.status_bar_icons_padding_bottom)
         );
     }
-- 
2.34.1
